src/app/snapp_test_transaction/snapp_test_transaction.ml

```diff
@@ -575,7 +581,72 @@ let create_snapp_account =
        create_command ~keyfile ~fee ~snapp_keyfile ~amount ~nonce ~memo))
 
 let upgrade_snapp =
-  Command.(async ~summary:"" (Param.return (fun () -> Deferred.unit)))
+  let create_command ~keyfile ~fee ~nonce ~memo ~snapp_keyfile ~verification_key
+      () =
+    let open Deferred.Let_syntax in
+    let%bind keypair = Util.keypair_of_file keyfile in
+    let%bind snapp_account_keypair = Util.keypair_of_file snapp_keyfile in
+    let verification_key =
+      let data =
+        Side_loaded_verification_key.of_base58_check_exn verification_key
+      in
+      let hash = Snapp_account.digest_vk data in
+      Snapp_basic.Set_or_keep.Set { With_hash.data; hash }
+    in
+    let spec =
+      { Transaction_snark.For_tests.Spec.sender = (keypair, nonce)
+      ; fee
+      ; receivers = []
+      ; amount = Currency.Amount.zero
+      ; snapp_account_keypair
+      ; memo = Util.memo memo
+      ; new_snapp_account = false
+      ; snapp_update = { Party.Update.dummy with verification_key }
+      ; current_auth = Permissions.Auth_required.Proof
```

Comment:
Could you make this an input? We can change permissions one should be able to upgrade based on the permissions they chose